src/goImport.ts

```diff
@@ -48,10 +48,46 @@ export function addImport(arg: string) {
 				editBuilder.insert(new vscode.Position(closeParenLine, 0), '\t"' + imp + '"\n');
 			});
 		} else if (imports.length > 0) {
-			// There are only single import declarations, add after the last one
-			let lastSingleImport = imports[imports.length - 1].end;
+			// There are some number of single line imports, which can just be collapsed into a block import.
+			// NOTE: this most often happens when the go imports tool is run, adds a single import
+			// then later the user manually imports something via this command.
 			return vscode.window.activeTextEditor.edit(editBuilder => {
-				editBuilder.insert(new vscode.Position(lastSingleImport + 1, 0), 'import "' + imp + '"\n');
+				/* The import statement(s) probably look something like this right now:
+
+				import "apackage"
+				// A comment
+				import . "dotinclude"
+				import "apackage2" // A comment explaining the package
+
+				We want to convert this to:
+
+				import (
+					"apackage"
+					// A comment
+					. "dotinclude"
+					import "apackage2" // A comment explaining the package
+				)
+
+				The process to clean this up is to:
+				1. Add `import (\n` before the first single line import
+				2. Replace `import` with `\t` for every single line import
+				3. Insert the new import at the end of the imports
+				4. Add `)` to close the import block
+				5. The user then saves, and goimports will clean everything up
+				*/
+				let firstImport = imports[0];
+				let lastImport = imports[imports.length - 1];
+				const importLength = 'import'.length;
+
+				editBuilder.insert(new vscode.Position(firstImport.start, 0), 'import (\n');
+
+				imports.forEach(element => {
+					editBuilder.replace(new vscode.Range(new vscode.Position(element.start, 0), new vscode.Position(element.start, importLength)), '\t');
```

Comment:
Depending on `importLength` will cause issues when there are spaces/tabs before the word "import".
This shouldn't happen if formatting has been run on the file, but there might be cases where user has not yet saved the file or has turned formatOnSave off or has not formatted the doc at all